packages/salesforce-adapter/src/filters/remove_fields_and_values.ts

```diff
@@ -53,6 +53,9 @@ const removeValuesFromInstances = (
 
   elements
     .filter(isInstanceElement)
+    // The below filter is temporary optimization to save calling transformValues for all instances
+    // since TYPE_NAME_TO_FIELD_REMOVALS contains currently only top level types
+    .filter(inst => typeNameToFieldRemovals.get(metadataType(inst)) !== undefined)
```

Comment:
same can be done for the the loop removing fields from object types.

also:
```suggestion
    .filter(inst => typeNameToFieldRemovals.has(metadataType(inst)))
```

Reply:
changed it, for types it won't have any effect though.

Reply:
yes, you are right about the types
my mistake